src/PowerShell/PowerShellManager.cs

```diff
@@ -106,6 +123,27 @@ private void RegisterStreamEvents()
             _pwsh.Streams.Warning.DataAdding += streamHandler.WarningDataAdding;
         }
 
+        /// <summary>
+        /// Create the PowerShell function that is equivalent to the 'scriptFile' when possible.
+        /// </summary>
+        private void DeployAzFunctionToRunspace()
```

Comment:
The content of the `$MyInvocation` variable seems to be quite different before and after this change. Even though `$PSScriptRoot` and `$PSCommandPath` are identical, `$MyInvocation.MyCommand.Path` used to contain the path to the script, and now this field is absent. There are many other changes as well - I can provide more details if you need them.

This is a potentially breaking change for existing customers. If you decide to proceed with this optimization by default anyway, consider documenting this and/or providing an opt-out mechanism.

Reply:
Right, documentation needs to be updated. The issue #225 was opened to track the documentation updates.

The change in `$MyInvocation.MyCommand` is by design, it's was pointing to a `ExternalScriptInfo` object previously because we were calling `run.ps1` directly, and now it's pointing to a `FunctionInfo` object, which doesn't have the `Path` property,

It's a breaking change that should definitely be called out in the documentation. But I don't think we need to provide an opt-out mechanism, partially because we are still in public preview and partially because it's something easy to change for the user.